File: ComponentImpl/src/main/java/com/xiaojinzi/component/support/RouterRequestHelp.java
package com.xiaojinzi.component.support;

import androidx.annotation.NonNull;
import androidx.annotation.UiThread;

import com.xiaojinzi.component.impl.RouterRequest;

/**
 * 这是一个 {@link com.xiaojinzi.component.impl.RouterRequest} 对象的帮助类
 */
public class RouterRequestHelp {

    @UiThread
    public static void executeBeforeAction(@NonNull RouterRequest request) throws Exception {
        Utils.checkNullPointer(request, "request");
        if (request.beforeAction != null) {
            request.beforeAction.invoke();
        }
    }

    @UiThread
    public static void executeAfterAction(@NonNull RouterRequest request) throws Exception {
        Utils.checkNullPointer(request, "request");
        if (request.afterAction != null) {
            request.afterAction.invoke();
        }
        executeAfterEventAction(request);
    }

    @UiThread
    public static void executeAfterErrorAction(@NonNull RouterRequest request) throws Exception {
        Utils.checkNullPointer(request, "request");
        if (request.afterErrorAction != null) {
            request.afterErrorAction.invoke();
        }
        executeAfterEventAction(request);
    }

    @UiThread
    public static void executeAfterEventAction(@NonNull RouterRequest request) throws Exception {
        Utils.checkNullPointer(request, "request");
        if (request.afterEventAction != null) {
            request.afterEventAction.invoke();
        }
    }

}

File: ComponentImpl/src/main/java/com/xiaojinzi/component/support/RouterRequestHelp.kt
package com.xiaojinzi.component.support

import androidx.annotation.UiThread
import com.xiaojinzi.component.impl.RouterRequest

/**
 * 这是一个 [com.xiaojinzi.component.impl.RouterRequest] 对象的帮助类
 */
object RouterRequestHelp {

    @UiThread
    @Throws(Exception::class)
    fun executeBeforeAction(request: RouterRequest) {
        request.beforeAction?.invoke()
    }

    @UiThread
    @Throws(Exception::class)
    fun executeAfterAction(request: RouterRequest) {
        request.afterAction?.invoke()
        executeAfterEventAction(request)
    }

    @UiThread
    @Throws(Exception::class)
    fun executeAfterErrorAction(request: RouterRequest) {
        request.afterErrorAction?.invoke()
        executeAfterEventAction(request)
    }

    @UiThread
    @Throws(Exception::class)
    fun executeAfterEventAction(request: RouterRequest) {
        request.afterEventAction?.invoke()
    }

}